Rename createdAt and updatedAt columns in users table to follow snake…

ujwalkumar1995 opened the pull request:
Closes #100 

# Description
Renamed createdAt and updatedAt column in user table to follow snake_case

## Testing

Can run tests under server folder. It has tests to validate migrations. 

## Type of change
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots
N/A

# Checklist:

- [x] Changes have new/updated automated tests, if applicable
- [x] Changes have new/updated docs, if applicable
- [x] I have performed a self-review of my own code
- [x] I have added comments on any new, hard-to-understand code
- [x] Changes have been manually tested
- [x] Changes generate no new errors or warnings

devonzara commented on server/migrations/20230521192838-rename-createdAt-and-updatedAt-in-user.js:
```diff
@@ -0,0 +1,14 @@
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up (queryInterface, Sequelize) {
```
lint: remove unused `Sequelize`

devonzara commented on server/migrations/20230521192838-rename-createdAt-and-updatedAt-in-user.js:
```diff
@@ -0,0 +1,14 @@
+    await queryInterface.renameColumn('users', 'updatedAt', 'updated_at');
+  },
+
+  async down (queryInterface, Sequelize) {
```
lint: remove unused `Sequelize`

devonzara commented on server/migrations/20230521192838-rename-createdAt-and-updatedAt-in-user.js:
```diff
@@ -0,0 +1,14 @@
+  async down (queryInterface, Sequelize) {
+    await queryInterface.renameColumn('users', 'created_at', 'createdAt');
+    await queryInterface.renameColumn('users', 'updated_at', 'updatedAt');
+  }
```
lint: add missing dangling comma

devonzara commented on server/models/User.model.ts:
```diff
@@ -71,9 +71,11 @@ class User extends Model<UserAttributes, UserCreationAttributes> {
   website: string;
 
   @CreatedAt
+  @Column({ field: "created_at" })
```
Instead of manually overriding the names, I'd probably opt to use the [`underscored`](https://sequelize.org/docs/v6/other-topics/naming-strategies/#the-underscored-option) option when initializing Sequelize.

devonzara commented on server/models/User.model.ts:
```diff
@@ -71,9 +71,11 @@ class User extends Model<UserAttributes, UserCreationAttributes> {
 
   @CreatedAt
+  @Column({ field: "created_at" })
   createdAt: Date;
```
I'd also recommend changing this to match the expected naming convention: `created_at`

devonzara commented on server/models/User.model.ts:
```diff
@@ -71,9 +71,11 @@ class User extends Model<UserAttributes, UserCreationAttributes> {
   createdAt: Date;
 
   @UpdatedAt
+  @Column({ field: "updated_at"})
   updatedAt: Date;
```
Same as the two previous comments...

devonzara reviewed (changes requested):
Welcome, and thanks for joining the project! Are you in our Discord thread? If so, please shoot @timmyichen a message to be added as a code owner. We typically don't work with forks in this repo as it's ran/organized more like a work/team environment.

In regards to the PR... looks like a good start! The changes you've made will certainly work for renaming the database fields, but there's a few lingering concerns:
1. This change doesn't change the name of the model's attributes, so we'd still have to reference it using camelCase naming.
2. We'd have to remember to do this to each new model we create in the future, it can be simplified by setting `underscored: true` in Sequelize's options.

In addition to the suggested changes, be sure to update the names in the type definitions found in `shared/User.ts` and `server/models/User.model.ts`.

Lastly, please remember to run `npm run lint`, `npm run test:server`, and `cypress run` (the last one requires the dev server be running) before pushing.

ujwalkumar1995 commented:
Hey @devonzara 
Thank you for the feedback.
I have incorporated the changes that you suggested.
I did run the lint and the tests. I was not quite able to run the e2e tests. Was facing some issues there. I am assuming they do run as part of the CI, so if they fail due to my changes, please do let me know I would be happy to fix the breaking change.
One more heads up regarding the use of underscored: true, as per my understanding on the docs we cannot use it as a global setting on sequelize but have to use it on each model. So for now I have added this property to the user model.

Please let me know if anything else needs to be fixed here.

ujwalkumar1995 commented:
Hey @devonzara 
I did double check this locally and also read couple of github issues regarding it. It is supported on the global config for sequelize. So have made the change accordingly.
Anyways, thank you for the knowledge share.

devonzara reviewed (approved):
Looks good to me... thanks again for taking this on!

ujwalkumar1995 commented:
Hey @timmyichen 
Could you please review this and let me know in case any changes are required?

timmyichen reviewed (approved):
Thanks for taking this on! Feel free to merge at your leisure :)
